### PR TITLE
Add ruok metric

### PR DESCRIPTION
**Pull request by aeuio**
zk_service_health is the new metric - add this to all stats collectd.

### CODE REVIEW

**Review by mpetazzoni — CHANGES_REQUESTED** [review]
I would have the `ZooKeeperServer` class make both `ruok` and `mntr` checks in its `get_stats()` method, instead of instantiating two of them only to have it do some special casing to call `_parse_ok()` or `_parse()`.

**Comment by aeuio** [comment]
Fixed!

File: zk-collectd.py [hunk 1]
```diff
@@ -43,30 +44,55 @@ def __init__(self, host='localhost', port='2181', timeout=1):
 
     def get_stats(self):
         """Get ZooKeeper server stats as a map."""
-        data = self._send_cmd('mntr')
-        return self._parse(data)
+        stats = {}
+        for cmd in ZK_CMDS:
+            data = self._send_cmd(cmd)
+            if cmd == "ruok":
+                results = self._parse_ok(data)
```
Comment (mpetazzoni) [inline zk-collectd.py]: So, again, if you have to special case here inside a loop, it makes using a loop somewhat meaningless. Just do:

``` python
def get_stats(self):
    stats = {}
    stats.update(self._get_health_stats())
    stats.update(self._get_mntr_stats())
    # We can add more metrics here in the future.
    return stats

def _get_health_stat(self):
    response = self._send_cmd(RUOK_CMD)
    return {'zk_service_health': int(response == IMOK_RESP)}

def _get_mntr_stats(self):
    response = self._send_cmd(MNTR_CMD)
    result = {}
    for line in response.splitlines():
        try:
            key, value = self._parse_line(line)
            if key == 'zk_server_state':
                result['zk_is_leader'] = int(value != 'follower')
            elif key == 'zk_version':
                continue
            else:
                result[key] = value
    return result
```

File: zk-collectd.py [hunk 2]
```diff
@@ -99,30 +125,24 @@ def _parse_line(self, line):
...
-                        val.type = 'counter' if k in COUNTERS else 'gauge'
-                        val.type_instance = k
-                        val.values = [v]
-                        val.plugin_instance = conf['instance']
-                        val.dispatch()
-                    except (TypeError, ValueError):
-                        collectd.error(('error dispatching stat; host=%s, '
-                                        'key=%s, val=%s') % (host, k, v))
-                        pass
-            except socket.error:
```
Comment (mpetazzoni) [inline zk-collectd.py:120]: Is it safe to let this error bubble up? Doesn't collectd automatically disable a plugin if it errors out too much?

**Review by mpetazzoni — CHANGES_REQUESTED** [review]

File: zk-collectd.py (hunk 2 above)
Comment (aeuio) [reply zk-collectd.py:120]: I replaced it with a log - it was there originally so i left it alone

File: zk-collectd.py (hunk 1 above)
Comment (aeuio) [reply zk-collectd.py]: Fixed it up!

**Comment by aeuio** [comment]
Thanks again!  Wasn't sure how much i should be changing anything because I was adding things so this was really helpful feedback.

**Review by mpetazzoni — APPROVED** [review]

**Comment by mpetazzoni** [comment]
LGTM. Can you confirm you've tested your latest changes?

**Comment by aeuio** [comment]
Thanks Max!  Yes, tested both success and failure conditions before i made the commit.

**Pull request merged**